src/lib/Form/Type/Location/LocationTrashType.php

```diff
@@ -57,15 +57,16 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 ['label' => /** @Desc("Send to trash") */ 'location_trash_form.trash']
             );
 
-        $builder->get('trash_options')->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event): void {
+        $builder->get('trash_options')->addEventListener(FormEvents::POST_SET_DATA, function (FormEvent $event): void {
             $form = $event->getForm();
+            $parentForm = $form->getParent();
             $this->trashTypeStrategy->addOptions(
                 $form,
-                $form->getParent()->getData()->getLocation()
+                $parentForm?->getData()->getLocation()
```

Comment:
This event listener should always be executed with parent form context available. If it's missing, then that is a logic error (i.e. `LogicException`, event listener is attached incorrectly).